### DISCUSSION TITLE
Missing wavesurfer methods in new version

### DISCUSSION THREAD

**Discussion by pokemon963** [katspaugh/wavesurfer.js#3069] [Q&A]
Hi! I have been using wavesurfer in my website and am trying to adapt my code for the new version released. There were some functions that I am getting errors for indicating that it is not an existing function.

Here is my WaveSurfer object:
```
wavesurfer = WaveSurfer.create({
      container: '#waveform',
      autoScroll: true,
      height: 80,
      // barWidth: 2,
      // barHeight: 1, // the height of the wave
      // barGap: null, // the optional spacing between bars of the wave, if not provided will be calculated in legacy format.
      waveColor: 'violet',
      progressColor: 'purple',
      normalize: true,
      plugins: [
        WaveSurfer.Timeline.create({}),
        WaveSurfer.Hover.create({
          lineColor: '#000',
          lineWidth: 3,
          labelBackground: '#555',
          labelColor: '#fff',
          labelSize: '10px',
        }),
        WaveSurfer.Regions.create({})
      ]
  });
```

Here are the functions I am getting errors for: 
`wavesurfer.setDisabledEventEmissions(['seek']);`, `wavesurfer.clearRegions();`, `wavesurfer.addRegion({ start: parseFloat(textfile_list[0][0]), end: parseFloat(textfile_list[1][0]), drag: false, color: 'hsla(39, 100%, 50%, 0.2)' });`

Any help I can get with this will be greatly appreciated! I am struggling to find specific documentation on the available methods.

**Comment by katspaugh** [comment]
Region-related methods are now called not on the wavesurfer object but on the Regions plugin itself. See this example: https://wavesurfer-js.org/examples/#regions.js


`setDisabledEventEmissions` won’t be replaced. What was your use case for it?

**Reply by pokemon963** [reply]
Thank you! I will do that.

With regards to `setDisabledEventEmissions`, it is related to this issue I have sent before https://github.com/katspaugh/wavesurfer.js/issues/2652#issuecomment-1416391491

**Reply by pokemon963** [reply]
Hi @katspaugh, may I ask where the documentation is for `setDisabledEventEmissions`, I am still getting the error that `wavesurfer.setDisabledEventEmissions` does not exist. So I am assuming I am using the function incorrectly.

Also, has the seeking event changed from before? Does `wavesurfer.skip` not set off the event?

**Reply by katspaugh** [reply]
The docs are here: https://wavesurfer-js.org/docs/classes/wavesurfer.default
But you won't find `wavesurfer.setDisabledEventEmissions` there because the method isn't there in v7.
You can unsubscribe and resubscribe to events in your application code, or use a boolean flag to ignore events in their handlers.

> Also, has the seeking event changed from before? Does wavesurfer.skip not set off the event?

The `seeking` event _is_ emitted when calling `skip`. See [this example](https://wavesurfer-js.org/examples/#a792a946e098cc97173af2ed08ff040c). Cheers.

**Comment by jlian2** [comment]
https://wavesurfer-js.org/examples/#regions.js

This example doe not work for me. Can you double check your source code? 

Everything is undefined in your js

**Comment by jlian2** [comment]
I have some other functions in script so I cannot import. However, if I do something like:
<script type="text/javascript" src="https://unpkg.com/wavesurfer.js@7/dist/plugins/regions.esm.js"></script>
<script type="text/javascript" src="https://unpkg.com/wavesurfer.js@7/dist/wavesurfer.esm.js"></script>

It does not work at all

**Reply by katspaugh** [reply]
You should read up on modules: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Modules

P.S. This is off topic, please use your own thread in the forum.

**Reply by jlian2** [reply]
Let me look at module. It turns out that <script> </script> does not work.

**Reply by jlian2** [reply]
Get it worked for modules. Thanks!
